server/tests/auth/test_integration_credentials_context_safe.py

```
@@ -0,0 +1,73 @@
+
+def _called_names(func: ast.FunctionDef) -> set[str]:
+    names: set[str] = set()
+    for node in ast.walk(func):
+        if isinstance(node, ast.Call) and isinstance(node.func, ast.Name):
+            names.add(node.func.id)
+    return names
+
+
+@pytest.mark.parametrize("rel_path,func_name", _CREDENTIAL_HELPERS)
+def test_credential_helper_is_request_context_safe(rel_path, func_name):
+    source = open(os.path.join(_SERVER_DIR, rel_path)).read()
```

Ruff (0.15.18)

[warning] Use a context manager for opening files

(SIM115)